packages/components/src/tools-panel/tools-panel/component.tsx

```diff
@@ -24,11 +24,14 @@ const ToolsPanel = (
 		panelContext,
 		resetAllItems,
 		toggleItem,
-		...toolsPanelProps
```

Comment:
I believe that removing this line can potentially introduce breaking changes, since we are not passing extra props to the wrapper component (e.g. any standard HTML props, like `data-*` attributes, `role`, `aria-*` props...)

We should probably reintroduce the spreading of those extra props to the `Grid` wrapper.

Reply:
Agreed, the removal of that was probably an overreach in response to the request to not expose any props on the `ToolsPanel` until they had been considered more.

To reinstate the collection, and subsequent spreading, of additional props for the `ToolsPanel` I ran into problems being able to appease TypeScript while attempting to exclude `Grid` props. If you have any ideas on the best way to achieve that, I'd appreciate your input.

In the meantime, I could easily clear any `Grid` props before they are passed through to the `ToolsPanel` component. I've included a diff below to illustrate what I'm meaning. If that happens to be acceptable, I'll create a PR for it.

<details>
  <summary>Patch to clear Grid props</summary>
  
```diff
diff --git a/packages/components/src/tools-panel/tools-panel/component.tsx b/packages/components/src/tools-panel/tools-panel/component.tsx
index 5492e19341..6525a62d3c 100644
--- a/packages/components/src/tools-panel/tools-panel/component.tsx
+++ b/packages/components/src/tools-panel/tools-panel/component.tsx
@@ -24,14 +24,14 @@ const ToolsPanel = (
 		panelContext,
 		resetAllItems,
 		toggleItem,
-		className,
+		...toolsPanelProps
 	} = useToolsPanel( props );
 
 	// Props are not directly passed through to avoid exposing Grid props
 	// until agreement has been reached on how ToolsPanel layout should be
 	// handled.
 	return (
-		<Grid columns={ 2 } className={ className } ref={ forwardedRef }>
+		<Grid { ...toolsPanelProps } columns={ 2 } ref={ forwardedRef }>
 			<ToolsPanelContext.Provider value={ panelContext }>
 				<ToolsPanelHeader
 					label={ label }
diff --git a/packages/components/src/tools-panel/tools-panel/hook.ts b/packages/components/src/tools-panel/tools-panel/hook.ts
index 2cedad4e73..2efd3b3121 100644
--- a/packages/components/src/tools-panel/tools-panel/hook.ts
+++ b/packages/components/src/tools-panel/tools-panel/hook.ts
@@ -216,5 +216,19 @@ export function useToolsPanel(
 		resetAllItems,
 		toggleItem,
 		className: classes,
+		// The following properties are cleared so as to prevent `Grid` props
+		// being passed through. This is intended to be removed once it is
+		// decided on exactly which `Grid` props the `ToolsPanel` should expose.
+		align: undefined,
+		alignment: undefined,
+		columns: undefined,
+		columnGap: undefined,
+		isInline: undefined,
+		gap: undefined,
+		justify: undefined,
+		rowGap: undefined,
+		rows: undefined,
+		templateColumns: undefined,
+		templateRows: undefined,
 	};
 }

```
</details>

Reply:
At the moment, the fact that `ToolsPanel` uses `Grid` internally is an implementation detail. We're not exposing that in any APIs, docs or types. Therefore, I don't think that we need to explicitly exclude `Grid`-related props, as we never included them in the first place.

Sure, that won't stop a developer from passing those props to `ToolsPanel` at runtime — but that's not the correct, documented usage anyway.

<details>

<summary>I think we should just bring back the spreading of `toolsPanelProps` without worrying too much about the rest:</summary>

```diff
diff --git a/packages/components/src/tools-panel/tools-panel/component.tsx b/packages/components/src/tools-panel/tools-panel/component.tsx
index 5492e19341..f7f16240ce 100644
--- a/packages/components/src/tools-panel/tools-panel/component.tsx
+++ b/packages/components/src/tools-panel/tools-panel/component.tsx
@@ -24,14 +24,11 @@ const ToolsPanel = (
 		panelContext,
 		resetAllItems,
 		toggleItem,
-		className,
+		...toolsPanelProps
 	} = useToolsPanel( props );
 
-	// Props are not directly passed through to avoid exposing Grid props
-	// until agreement has been reached on how ToolsPanel layout should be
-	// handled.
 	return (
-		<Grid columns={ 2 } className={ className } ref={ forwardedRef }>
+		<Grid { ...toolsPanelProps } columns={ 2 } ref={ forwardedRef }>
 			<ToolsPanelContext.Provider value={ panelContext }>
 				<ToolsPanelHeader
 					label={ label }
```
</details>

Reply:
A PR to apply the above changes can be found in: https://github.com/WordPress/gutenberg/pull/36428